src/mlfmu/fmu_build/templates/onnx_fmu/onnxFmu.cpp

```diff
@@ -55,7 +55,26 @@ std::wstring OnnxFmu::formatOnnxPath(cppfmu::FMIString fmuResourceLocation)
  */
 void OnnxFmu::CreateSession()
 {
-    session_ = Ort::Session(env, onnxPath_.c_str(), Ort::SessionOptions {nullptr});
+    #ifdef _WIN32
+        // Windows
+        // Ort::Session::Session(const Ort::Env &, const wchar_t *,const Ort::SessionOptions &)
+        session_ = Ort::Session(env, onnxPath_.c_str(), Ort::SessionOptions {nullptr});
+    #else
+        // Linux
+        // Ort::Session::Session(Ort::Env&, const char*, Ort::SessionOptions)
+        const wchar_t * input = onnxPath_.c_str();
+        size_t size = (wcslen(input) + 1) * sizeof(wchar_t);
+        char * buffer = new char[size];
+        #ifdef __STDC_LIB_EXT1__
+            // wcstombs_s is only guaranteed to be available if __STDC_LIB_EXT1__ is defined
+            size_t convertedSize;
+            std::wcstombs_s(&convertedSize, buffer, size, input, size);
+        #else
+            std::wcstombs(buffer, input, size);
+        #endif
+        session_ = Ort::Session(env, buffer, Ort::SessionOptions {nullptr});
+        delete[] buffer;
+    #endif
```

Comment:
Good work finding this out! It looks like this was not easy to figure out what would work for linux. 

To me it maybe looks like the problem stems from Windows needs the path to be a `wchar_t *` while Linux needs it to be a `char *`. 

My only question is then if this is better handled in the `formatOnnxPath` function and save in the correct format in the variable `onnxPath_`. Then we can just save a `std::wstring` in Windows and a `std::string` in Linux. Then I think we can just call `.c_str()` in both cases? 

However, I do not know how easy it is to convert from a `std::wstring` to a `std::string`? I remembered working with this was a hassle when I first wrote it

Reply:
Asking copilot for a quick suggestion it suggests. But I do not know what will work.

For older version than C++17
```cpp
std::string wstring_to_string(const std::wstring& wstr) {
    std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
    return converter.to_bytes(wstr);
}
```

For newer versions:
```cpp
std::string wstring_to_string(const std::wstring& wstr) {
    return std::string(wstr.begin(), wstr.end());
}
```

Reply:
Great points! I got the suggestion w.r.t. conversion from StackOverflow, but don't remember date of that article. The tricky part is w.r.t. needing to put in a wchar_t * for Windows and a char * for Linux (non-Windows - to be tested on Mac), which is why I put it in this location.  It is being stored as a class var as a std::wstring. So then we would wanna also put a macro around the class var for the right type in Windows vs non-Windows. Could try the method you found to go from wstring to string and then pass as char * - would need to test at home again.
I don't know what is necessarily 'cleaner' or better, but the suggestions you have are fewer lines of code at least (if they work) :)

Reply:
Updated. I changed the func to not return a value, since I saw no reason for that, given that you anyway store to a class variable. See if you think this looks ok :)

Reply:
The only reason for the function to return the value instead of saving it to the class variable in the function is for it to be easier to test and easier to reuse the function. But in this case I agree it is good to do as you did. 

Getting the function to return a different type depending on the OS in C++ does not sound like the easiest thing to get right. And there is probably no reason to reuse this function at a later point anyway. 

So I agree

Reply:
Ok, yes, could do that also with macros, I just did not see the point since we aren't using it elsewhere and I did not expect we would either. Can always be changed later :)